api/controllers/measurements.js

```diff
@@ -101,29 +101,49 @@ module.exports.query = function (query, page, limit, cb) {
   }
 
   //
-  // Handle custom sorts, starting with default of most recent measurements
-  // first. Do nothing if we don't have both sort and order_by.
+  // Handle custom sorts, starting with default of most recent measurements first.
   //
-  var sort = { column: 'date_utc', direction: 'desc' };
-  if (_.has(payload, 'sort') && _.has(payload, 'order_by')) {
-    // Catch case where order_by is provided as 'date'
-    if (payload.order_by === 'date') { payload.order_by = 'date_utc'; }
+  const defaultSort = [{ column: 'date_utc', direction: 'desc' }];
+  let sort = [];
+  if (_.has(payload, 'order_by')) {
+    payload.order_by = [].concat(payload.order_by);
+    if (_.has(payload, 'sort')) {
+      payload.sort = [].concat(payload.sort);
+    }
+    payload.order_by.forEach((column, i, columns) => {
+      // skip unknown columns
+      if (projection.indexOf(column) === -1) { return; }
+
+      // Catch case where order_by is provided as 'date'
+      if (column === 'date') {
+        column = 'date_utc';
+      }
+      if (column === 'sourceName') {
+        column = 'source_name';
+      }
+      let direction;
+      try {
+        direction = payload.sort[i];
+      } catch (err) {
+        direction = 'asc';
```

Comment:
If there's no `order_by` in the query, it'll default to `date_utc` with `desc`. Latest first.

Otherwise, when there's some `order_by` filters, and no matching `sort`, it'll default to `asc`. I'm using `asc` as default sort direction for the others too.